feat: Add inline loading fragment component

MoizMasud opened the pull request:
closes https://github.com/IBM/carbon-ui-builder/issues/122

zvonimirfras committed:
Add UIText to fragment preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add UINumberInput to fragment preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

Commit:
Add inline loading component

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add vertical support to progress indicator export

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add UIProgressIndicator to fragment preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

Commit:
Lint alittle

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

Commit:
Lint alittle

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras commented on src/fragment-components/a-inline-loading.tsx:
```diff
@@ -0,0 +1,140 @@
+
+export const AInlineLoadingSettingsUI = ({ selectedComponent, setComponent }: any) => {
+	return <>
+		<legend className="bx--label">Status</legend>
```
Use single quotes everywhere. I think the only place we use double is in exports and that's because of wrapping quotes in quotes.

zvonimirfras commented on src/fragment-components/a-inline-loading.tsx:
```diff
@@ -0,0 +1,140 @@
+export const AInlineLoadingSettingsUI = ({ selectedComponent, setComponent }: any) => {
+	return <>
+		<legend className="bx--label">Status</legend>
+		<RadioButtonGroup
+			orientation="vertical"
+			name="status-radio-buttons"
+			valueSelected={selectedComponent.status}
+			onChange={(event: any) => setComponent({
+				...selectedComponent,
+				status: event
+			})} >
+			<RadioButton
+			id="inactive"
+			labelText="Inactive"
+			value="inactive"
+			/>
+			<RadioButton
+			id="active"
+			labelText="Active"
+			value="active"
+			/>
+			<RadioButton
+			id="finished"
+			labelText="Finished"
+			value="finished"
+			/>
+			<RadioButton
+			id="error"
+			labelText="Error"
+			value="error"
+			/>
+		</RadioButtonGroup>
```
Follow the indentation pattern from the rest of the code base and put the closing `/>` on the same line.

zvonimirfras commented on src/fragment-components/a-inline-loading.tsx:
```diff
@@ -0,0 +1,140 @@
+	settingsUI: AInlineLoadingSettingsUI,
+	keywords: ['inline', 'loading'],
+	name: 'Inline Loading',
+	type: 'inlineloading',
```
```suggestion
	type: 'inline-loading',
```

and elsewhere where it's missing...

zvonimirfras commented on src/fragment-components/a-inline-loading.tsx:
```diff
@@ -0,0 +1,140 @@
+	name: 'Inline Loading',
+	type: 'inlineloading',
+	defaultComponentObj: {
+		type: 'inlineloading',
```
```suggestion
		type: 'inline-loading',
```

zvonimirfras commented on src/fragment-components/a-inline-loading.tsx:
```diff
@@ -0,0 +1,140 @@
+				return `<ibm-inline-loading
+							(onSuccess)="${nameStringToVariableString(json.codeContext?.name)}OnSuccess.emit($event)"
+							[state]="${nameStringToVariableString(json.codeContext?.name)}Status"
+							${json.status === 'active' ? `[loadingText]="'${json.textDescription}'"`: ''}
+							${json.status === 'finished' ? `[successText]="'${json.textDescription}'"`: ''}
+							${json.status === 'error' ? `[errorText]="'${json.textDescription}'"`: ''}
```
This won't work dynamically...

zvonimirfras replied:
Should there be 3 inputs in UI?

zvonimirfras commented on src/fragment-components/index.ts:
```diff
@@ -22,6 +22,7 @@ import * as textarea from './a-textarea';
 import * as textinput from './a-textinput';
 import * as link from './a-link';
 import * as loading from './a-loading';
+import * as inlineloading from './a-inline-loading';
```
```suggestion
import * as inlineLoading from './a-inline-loading';
```

etc

zvonimirfras commented on src/fragment-components/index.ts:
```diff
@@ -84,6 +86,7 @@ export const allComponents = {
 	fragment,
 	grid,
 	loading,
+	inlineloading,
```
```suggestion
	inlineLoading,
```

zvonimirfras commented on src/ui-fragment/src/components/ui-inline-loading.tsx:
```diff
@@ -0,0 +1,29 @@
+	setState: (state: any) => void;
+	setGlobalState: (state: any) => void;
+}) => {
+	if (state.type !== 'inlineloading') {
```
```suggestion
	if (state.type !== 'inline-loading') {
```

zvonimirfras commented on src/ui-fragment/src/utils.tsx:
```diff
@@ -65,6 +66,9 @@ export const renderComponents = (state: any, setState: (state: any) => void, set
 		case 'grid':
 			return <UIGrid key={state.id} state={state} setState={setState} setGlobalState={setGlobalState} />;
 
+		case 'inlineloading':
```
```suggestion
		case 'inline-loading':
```

Commit:
Fix indentation/changed double quote to single

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

Commit:
Add UI inputs for status

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras commented on src/fragment-components/a-inline-loading.tsx:
```diff
@@ -0,0 +1,139 @@
+export const AInlineLoadingSettingsUI = ({ selectedComponent, setComponent }: any) => {
+	return <>
+		<legend className='bx--label'>Status</legend>
+		<RadioButtonGroup
+			orientation='vertical'
+			name='status-radio-buttons'
+			valueSelected={selectedComponent.status}
+			onChange={(event: any) => setComponent({
+				...selectedComponent,
+				status: event
+			})} >
+			<RadioButton
+				id='inactive'
+				labelText='Inactive'
+				value='inactive'/>
+			<RadioButton
+				id='active'
+				labelText='Active'
+				value='active'/>
+			<RadioButton
+				id='finished'
+				labelText='Finished'
+				value='finished'/>
+			<RadioButton
+				id='error'
+				labelText='Error'
+				value='error'/>
+		</RadioButtonGroup>
```
```suggestion
		<RadioButtonGroup
		orientation='vertical'
		name='status-radio-buttons'
		valueSelected={selectedComponent.status}
		onChange={(event: any) => setComponent({
			...selectedComponent,
			status: event
		})}>
			<RadioButton
				id='inactive'
				labelText='Inactive'
				value='inactive' />
			<RadioButton
				id='active'
				labelText='Active'
				value='active' />
			<RadioButton
				id='finished'
				labelText='Finished'
				value='finished' />
			<RadioButton
				id='error'
				labelText='Error'
				value='error' />
		</RadioButtonGroup>
```

sigh

zvonimirfras committed:
Update src/fragment-components/a-inline-loading.tsx

zvonimirfras reviewed (changes requested):
Pros and cons of having all the angular inputs available in UI and then worked around in react export?

MoizMasud commented on src/fragment-components/a-inline-loading.tsx:
```diff
@@ -0,0 +1,139 @@
+				return `<ibm-inline-loading
+							(onSuccess)="${nameStringToVariableString(json.codeContext?.name)}OnSuccess.emit($event)"
+							[state]="${nameStringToVariableString(json.codeContext?.name)}Status"
+							${json.status === 'active' ? `[loadingText]="${nameStringToVariableString(json.codeContext?.name)}LoadingText"`: ''}
+							${json.status === 'finished' ? `[successText]="${nameStringToVariableString(json.codeContext?.name)}SuccessText"`: ''}
+							${json.status === 'error' ? `[errorText]="${nameStringToVariableString(json.codeContext?.name)}ErrorText"`: ''}
```
angular requires a different input for each type of state. where as react uses `status` input and `description` to take care of state and text of each state.

zvonimirfras replied:
Sure... Doing this will make it impossible to have any other state except the one it's in at the time of code generation.

zvonimirfras committed:
Add UISearchInput to fragment preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add UITag to fragment preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Rename textarea -> text-area

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add UITextArea to fragment preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Rename textinput -> text-input

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add UITextInput to fragment preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add missing `type` property to overflow-menu-item

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add UIOverflowMenu to fragment preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add UITileFold to fragment preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add UITile to fragment preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add UIClickableTile to fragment preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Fix toggle label text in export

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Use toggle code name as id

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add UIToggle to fragment preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add UIExpandableTile to fragment preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add UISelectableTile to fragment preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add UISelectableTileGroup to fragment preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add UIRadioTileGroup to fragment preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Update width height

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Enable fragment preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add ids and labels to some input components

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Improve accessibility

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Lint a little

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add some missing keys

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Fix div cannot be in p

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add missing props to combobox

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add missing ids to progress indicator

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Fix warnings on drag over

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add mising ids to radio tile group inputs

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add missing key to draggable list

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add some missing ids

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Show the landing page again

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Improve empty dashboard styling

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add an empty editor canvas placeholder

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Improve empty dashboard state

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Fix icon styling

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add feedback (github) and slack menu

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Put help menu last

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add icons to dashboard tile overflow menu

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add icons to header menus

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Stop using deprecated placeHolderText prop

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add fragment tile menu icon description

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Use id from fragment instead of params

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Refactor a bit

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add fragment preview modal

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add featured fragments dashboard (from github)

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Refactor modal system

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Reorganize `filenameToLanguage`

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add launch page

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Improve introduction page

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Improve help page style

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Move style helper functions to library

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Include styles in preview

closes #141

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Fix UIComboBox itemToString

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Fix UIComboBox titleText/label

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Fix input name when duplicating a component

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Move "Fork on GitHub" to Help menu

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Export all contexts in index

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add ContextProviders

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add GithubContext

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Use GithubContext instead of octokit directly

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Use github token with octokit

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Fix modal submit

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add login with github modal

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Update github login modal when token changes

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add logout github modal

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add user context

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Activate login with github

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Remove ; from all modals

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add repo page

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add getRepos to github context

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add all repos to user menu

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Show github username if full name not available

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Use githubLogin instead of name

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add github file preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Use GithubFilePreview in launch

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add github navigator component

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Use github navigator on repo page

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Generalize getRepos

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Generalize github navigator

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Use github navigator for launch

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Enable viewing repos per owner/org in launch

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add toolbar with "copy sharable link" button

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Hide toolbar on launch page

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Restyle github navigator

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add breadcrumbs on repo page

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Sort folder content by default

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add getContentWithFolder

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add file preview navigation arrows

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Fix disabled arrows for fragment previews (repo)

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Preview files regardles of case (upper, lower)

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add FAQ link to help

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Lint a little

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add notes section

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Remove someone's debugging leftover

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Hide code pane for now

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Change cursor to indicate draggable and clickable content

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

MoizMasud committed:
Fix svg title names (#150)

Co-authored-by: Moiz Masud <moiz.masud@ibm.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

MoizMasud committed:
Fix code snippet name (#144)

Co-authored-by: Moiz Masud <moiz.masud@ibm.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Externalize markdown styles

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add markdown preview to github file preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Allow "launching" public pages without logging in

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

Commit:
Fix indentation/angular export code

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

Commit:
Lint alittle

zvonimirfras committed:
Center image preview in github navigator

zvonimirfras committed:
Center image preview in github navigator

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Remove not needed scroll bar from image preview

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Move center style to index.scss

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add a loading indicator when switching to featured

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

MoizMasud committed:
Fix not exporting empty css classes (#151)

* Fix export scss classes with no content

* Added empty line back in

* Removed empty space

* Removed empty line

* Lint alittle

* Changed code to guards in function

Co-authored-by: Moiz Masud <moiz.masud@ibm.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

MoizMasud committed:
Add size to button (#149)

* Add size to button

* Add size to button preview

* Update src/fragment-components/a-button.tsx

Co-authored-by: Zvonimir Fras <zvonimir.fras@gmail.com>

* Code clean up

Co-authored-by: Moiz Masud <moiz.masud@ibm.com>
Co-authored-by: Zvonimir Fras <zvonimir.fras@gmail.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Update outline icons

zvonimirfras committed:
Update outline icons

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Set tooltips for toolbar icon buttons

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Make last modified date non-italic in header

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Ignore .vscode

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Lock buffer version

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Hide monaco typescript false positives validation

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Make editors that don't change anything read only

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Ignore case in user search string

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Keep LayoutWidget closed on page load

- prevents most crashes

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

Rafajrg21 committed:
Fix import JSON modal from dashboard tile (#164)

Signed-off-by: Rafael Romero <rafajrg21@gmail.com>

Signed-off-by: Rafael Romero <rafajrg21@gmail.com>
Co-authored-by: Rafael Romero <rromero@zeroq.cl>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

miqh committed:
Add cross-env for cross platform project start (#167)

Signed-off-by: Michael Huynh <miqid@outlook.com>

Signed-off-by: Michael Huynh <miqid@outlook.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

prasoonmhwr committed:
fix: "Toggle" commponent react export doesn't update state  (#165)

* bug fix 135

Signed-off-by: Prasoon Mahawar <prasoonmahawar21@gmail.com>

* signing DCO Signed-off-by: Author Name <prasoonmhwr21@gmail.com>

Signed-off-by: Prasoon Mahawar <prasoonmahawar21@gmail.com>

* resolved event value bug Signed-off-by: Author Name <prasoonmhwr21@gmail.com>

Signed-off-by: Prasoon Mahawar <prasoonmahawar21@gmail.com>

Signed-off-by: Prasoon Mahawar <prasoonmahawar21@gmail.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Update package-lock for cross-env

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Simplify generated component ids

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

miqh committed:
Add missing export attributes for tag component (#157) (#166)

Signed-off-by: Michael Huynh <miqid@outlook.com>

Signed-off-by: Michael Huynh <miqid@outlook.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

Dexters-Hub committed:
Fix Notification Component to make it to the front (#169)

* Fix Notification Component to make it to the front

Signed-off-by: alwin <alwinjohn231@gmail.com>

* changed z-index to 4 and position to absolute

Signed-off-by: alwin <alwinjohn231@gmail.com>

Signed-off-by: alwin <alwinjohn231@gmail.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

sadiejay committed:
Add issue forms (#176)

Signed-off-by: Sadie <19538219+sadiejay@users.noreply.github.com>

Signed-off-by: Sadie <19538219+sadiejay@users.noreply.github.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

sadiejay committed:
Add PR template feature (#177)

* Creates pull request template

Signed-off-by: Sadie <19538219+sadiejay@users.noreply.github.com>

* Fixes typo and reformats comment

Signed-off-by: Sadie <19538219+sadiejay@users.noreply.github.com>

Signed-off-by: Sadie <19538219+sadiejay@users.noreply.github.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Improve edit title tooltip

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Fix copy sharable link alignment

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Fix style sync issue when editing

react and angular export also fixed
json requires more foundational work

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Fix preview rendering

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Fix css json export

- remove the content because it's fetched from global
- content will be exported at the fragment level

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Fix chrome bug

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Stop using context in UI library

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Export full fragment in JSON

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add isFragment

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Use allCssClasses from state

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Detect componentObj as a fragment

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Make UICheckbox a controlled component

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Make UITextInput a controlled component

also fix imaginaryTarget issue

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Pass allCssClasses to UIFragment (view)

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Add support for full fragments' json in featured

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Use full fragment object in fragment preview modal

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Try catching layout widget crash

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Import css classes from json

- overrides existing classes

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Hide layout widget drag preview when not dragging

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

MoizMasud committed:
Removed empty line

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

MoizMasud committed:
Add additonal code

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

MoizMasud committed:
Lint alittle

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

MoizMasud committed:
Add additional code to react export

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

MoizMasud committed:
Clean up code

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

Akshat55 committed:
fix: Reference type instead of object key in additional code export (#193)

Signed-off-by: Akshat Patel <akshat@live.ca>

Signed-off-by: Akshat Patel <akshat@live.ca>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

Akshat55 committed:
ci: Add build workflow (#192)

* Add build workflow

Signed-off-by: Akshat Patel <akshat@live.ca>

* ci: Run deploy job only on push events to main branch

Signed-off-by: Akshat Patel <akshat@live.ca>

Signed-off-by: Akshat Patel <akshat@live.ca>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

MoizMasud committed:
Changed tooltip position to button (#196)

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

Akshat55 commented on src/fragment-components/a-inline-loading.tsx:
```diff
@@ -0,0 +1,238 @@
+			code: ({ json }) => {
+				const status = `state["${nameStringToVariableString(json.codeContext?.name)}"] || "${json.status}"`;
+				return `<InlineLoading
+					onSuccess={(success) => handleInputChange({
+						target: {
+							name: "${json.codeContext?.name}",
+						}
+					})}
```
@zvonimirfras 

I know for a fact this isn't correct, the idea behind on success is to call a function when status is `finished`. But the users are controlling the status, so I don't see a reason in having this onSuccess function at all, should we just remove it or do a check where we expect users will be passing in a function to execute on `finished`?

Other then this one change, this PR looks good to me, let me know what you think we should do here.

MoizMasud committed:
Change label for css class name (#200)

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

MoizMasud committed:
feat: Add missing Accordion attributes for angular (#195)

* Updated carbon components react export package version

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

* Updated disabled

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

* Changed disabled input in angular export

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

MoizMasud committed:
feat: Add missing features to Search component (#197)

* Add missing properties to search component

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

* Add props to ui-search preview file

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

* Lint fix

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

* Lint fix

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

* Update src/ui-fragment/src/components/ui-serach-input.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

* Update src/fragment-components/a-searchinput.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

* Update src/fragment-components/a-searchinput.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

* Update src/fragment-components/a-searchinput.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

* Update src/fragment-components/a-searchinput.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

* Update src/fragment-components/a-searchinput.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

* Update src/fragment-components/a-searchinput.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

* Update src/fragment-components/a-searchinput.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

* Update src/ui-fragment/src/components/ui-serach-input.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

* Change auto complete naming convention

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

* Update src/fragment-components/a-searchinput.tsx

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>
Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras commented on src/fragment-components/a-inline-loading.tsx:
```diff
@@ -0,0 +1,238 @@
+		onChange={(event: any) => setComponent({
+			...selectedComponent,
+			status: event
+		})} >
```
```suggestion
		})}>
```

zvonimirfras commented on src/fragment-components/a-inline-loading.tsx:
```diff
@@ -0,0 +1,238 @@
+			description: componentObj.successText
+		}
+	};
+	return (
```
```suggestion

	return (
```

zvonimirfras committed:
Update src/fragment-components/a-inline-loading.tsx

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

MoizMasud committed:
feat: Add content switcher component (#142)

* Add content-switcher

* Add UI-content-switcher file

* Add exports for content-switcher

* Lint alittle

* Lint alittle

* Add svg for content-switcher

* Fix indentation

* Lint alittle

* Update src/fragment-components/a-content-switcher.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

* Update src/fragment-components/a-content-switcher.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

* Add notice for size input on angular code export

* Add size prop to angular export

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

* Update carbon-components-angular package version

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

* Update content-switcher.svg

* Update content-switcher.svg

* Update src/fragment-components/a-content-switcher.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

* Update src/fragment-components/a-content-switcher.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

* Change updateItem function name

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>
Co-authored-by: Moiz Masud <moiz.masud@ibm.com>
Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

MoizMasud committed:
Add successDelay prop

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

zvonimirfras committed:
Fix elements search

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

Akshat55 commented on src/fragment-components/a-inline-loading.tsx:
```diff
@@ -0,0 +1,255 @@
+			code: ({ json }) => {
+				const status = `state["${nameStringToVariableString(json.codeContext?.name)}"] || "${json.status}"`;
+				return `<InlineLoading
+					onSuccess={(success) => handleInputChange({
+						target: {
+							name: "${json.codeContext?.name}",
+						}
+					})}
```
```suggestion
					onSuccess={() => {
						if(typeof state.${nameStringToVariableString(json.codeContext?.name)}OnSuccess === "function") {
							return state.${nameStringToVariableString(json.codeContext?.name)}OnSuccess;
						}
						return undefined;
					})}
```

Akshat55 replied:
If a function is passed by the user then we return the function otherwise undefined

MoizMasud committed:
Update src/fragment-components/a-inline-loading.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

MoizMasud committed:
Merge branch 'main' of https://github.com/MoizMasud/carbon-components-builder into inlineLoading

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

MoizMasud committed:
Update github file

Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>
